### PR TITLE
gh-95144: Improve error message of `... in None`

### PR DESCRIPTION
**Pull request by zware**
Add "container" to the message as a breadcrumb for looking up what's wrong.

* Issue: gh-95144

### CODE REVIEW

File: Objects/abstract.c [hunk 1]
```diff
@@ -2190,7 +2190,15 @@ _PySequence_IterSearch(PyObject *seq, PyObject *obj, int operation)
     it = PyObject_GetIter(seq);
     if (it == NULL) {
         if (PyErr_ExceptionMatches(PyExc_TypeError)) {
-            type_error("argument of type '%.200s' is not iterable", seq);
+            if (operation == PY_ITERSEARCH_CONTAINS) {
+                type_error(
+                    "argument of type '%.200s' is not a container or iterable",
```
Comment (erlend-aasland) [inline Objects/abstract.c:2163]:
```suggestion
                    "argument of type '%T' is not a container or iterable",
```

File: Objects/abstract.c [hunk 2]
```diff
@@ -2190,7 +2190,15 @@ _PySequence_IterSearch(PyObject *seq, PyObject *obj, int operation)
...
+                    seq
+                    );
+            }
+            else {
+                type_error("argument of type '%.200s' is not iterable", seq);
```
Comment (erlend-aasland) [inline Objects/abstract.c:2168]:
```suggestion
                type_error("argument of type '%T' is not iterable", seq);
```

File: Objects/abstract.c (hunk 1 above)
Comment (erlend-aasland) [reply Objects/abstract.c:2163]: PEP-737 syntax :)

Comment (zware) [reply Objects/abstract.c:2163]: Done, thanks :)

File: Objects/abstract.c [hunk 3]
```diff
@@ -2190,7 +2190,15 @@ _PySequence_IterSearch(PyObject *seq, PyObject *obj, int operation)
     it = PyObject_GetIter(seq);
     if (it == NULL) {
         if (PyErr_ExceptionMatches(PyExc_TypeError)) {
-            type_error("argument of type '%.200s' is not iterable", seq);
+            if (operation == PY_ITERSEARCH_CONTAINS) {
+                type_error(
+                    "argument of type '%T' is not a container or iterable",
```
Comment (erlend-aasland) [inline Objects/abstract.c]: Ah, `type_error` already looks up the type, so we should use `%N`.

```suggestion
                    "argument of type '%N' is not a container or iterable",
```

File: Objects/abstract.c [hunk 4]
```diff
@@ -2190,7 +2190,15 @@ _PySequence_IterSearch(PyObject *seq, PyObject *obj, int operation)
...
+                    seq
+                    );
+            }
+            else {
+                type_error("argument of type '%T' is not iterable", seq);
```
Comment (erlend-aasland) [inline Objects/abstract.c]:
```suggestion
                type_error("argument of type '%N' is not iterable", seq);
```

**Comment by erlend-aasland** [comment]
Sorry for the mess! We can't use PEP-737 formatting here; if so, `type_error` must be changed to _not_ access `tp_name`, and all its callers must change :) Sorry!

**Comment by zware** [comment]
@Mariatta, is this change enough to address the issue (gh-95144) in your opinion?

**Pull request merged**